src/Microsoft.ML.Data/Transforms/Normalizer.cs

```diff
@@ -387,32 +388,70 @@ internal static DataViewType LoadType(ModelLoadContext ctx)
                 Contracts.AssertValue(ctx);
                 // *** Binary format ***
                 //   - bool: is vector
-                //   - int: vector size
-                //   - byte: ItemKind of input column (only R4 and R8 are valid)
                 bool isVector = ctx.Reader.ReadBoolean();
-                int vectorSize = ctx.Reader.ReadInt32();
-                Contracts.CheckDecode(vectorSize >= 0);
-                Contracts.CheckDecode(vectorSize > 0 || !isVector);
+
+                if (ctx.Header.ModelVerWritten < 0x00010002)
+                {
+                    // *** Previous Binary format ***
+                    //   - int: vector size
+                    //   - byte: ItemKind of input column (only R4 and R8 are valid)
+                    int vectorSize = ctx.Reader.ReadInt32();
+                    Contracts.CheckDecode(vectorSize >= 0);
+                    Contracts.CheckDecode(vectorSize > 0 || !isVector);
+                    InternalDataKind itemKindOld = (InternalDataKind)ctx.Reader.ReadByte();
+                    Contracts.CheckDecode(itemKindOld == InternalDataKind.R4 || itemKindOld == InternalDataKind.R8);
+                    var itemTypeOld = ColumnTypeExtensions.PrimitiveTypeFromKind(itemKindOld);
+                    return isVector ? (DataViewType)(new VectorDataViewType(itemTypeOld, vectorSize)) : itemTypeOld;
+                }
+
+                // *** New Binary format ***
+                //   - int: number of dimensions
+                //   - ints: as many as dimensions, each one represent the size of each dimension
+                //   - byte: ItemKind of input column (only R4 and R8 are valid)
```

Comment:
>byte [](start = 23, length = 4)

If you read (and write) this byte first, then in the `!isVector` case you can return `itemType` without reading anything else.

Reply:
I've done this. Please review if it was what you were expecting. Thanks!

Reply:
Yes! Thanks!

---
In reply to: [334615811](https://github.com/dotnet/machinelearning/pull/4321#discussion_r334615811) [](ancestors = 334615811)